code/processes/wdb.q

```diff
@@ -313,7 +313,17 @@ merge:{[dir;pt;tableinfo;mergelimits;hdbsettings]
 	      curr[0]:curr[0],select from get segment;
 	      curr[1]:curr[1],segment;		
       	$[islast or mergemaxrows < count curr[0];
-	        [.lg.o[`merge;"upserting ",(string count curr[0])," rows to ",string dest];
+	        
+                [.lg.o[`resort;"Checking that the contents of this subpartition conform"];
+                pattrtest:10h=type @[{@[x;y;`p#]}[curr[0];];.merge.getextrapartitiontype[tablename];{x}];
```

Comment:
Can you just make this error trap return a 1b or 0b

Reply:
Sorry actually I'm not sure I see what you mean. If the error trap is only called when the function {@[x;y;`p#]} fails, shouldn't it just be a 1b in there? Otherwise just the successful table is returned

Reply:
I think what Jonny is suggesting is make that lambda return e.g. 1b if it succeeds e.g. ``{@[x;y;`p#];1b}`` and then in the error trap return 0b if it fails, so you don't need the `10h=type` bit at the start, as you already have a boolean & don't need to do another check